File: Offer/2224/main.ts
function reverseList(head: ListNode | null): ListNode | null {
    let root = new ListNode()
    let temp: ListNode | null = null
    while (head != null){
        temp = head.next
        head.next = root.next
        root.next = head
        head = temp
    }
    return root.next
};

File: Offer/2224/main.go
package main

type ListNode struct {
	Val  int
	Next *ListNode
}

func reverseList(head *ListNode) *ListNode {
	if head == nil {
		return nil
	}

	var root *ListNode = &ListNode{0, nil}
	var temp *ListNode

	for head != nil {
		temp = head.Next
		head.Next = root.Next
		root.Next = head
		head = temp
	}
	return root.Next
}
